spec/src/main/asciidoc/query-language.asciidoc

```diff
@@ -324,6 +324,8 @@ The `set` clause, with syntax given by `set_clause`, specifies a list of updates
 
 The `order` clause (or `order by` clause), with syntax given by `orderby_clause`, specifies a lexicographic order for the query results, that is, a list of entity fields used to sort the records which satisfy the restriction imposed by the `where` clause. The keywords `asc` and `desc` specify that a given field should be sorted in ascending or descending order respectively; when neither is specified, ascending order is the default.
 
+By default, the order of values of a Java enumerated type is determined by the integer returned by `ordinal()`. A programming model for entity classes might allow control over the order of `enum` values. For example, Jakarta Persistence allows this via the `@Enumerated` annotation.
```

Comment:
Are you sure that the `ordinal` should be the default?

If you include a new element or swap it, it will generate issues with data integrity.

When comparing String and `ordinal`, the unique benefit that came to my mind about ordinal is performance once it is smaller.

But, readability, data integrity, and flexibility I would prefer as String.

At the NoSQL side, we mostly use the `name`method, some samples:

- Couchbase
- Amazon DynamoDB
- RethinkDB
- Elasticsearch
- Cassandra
- Redis

Reply:
This is an unfortunate situation where neither option is ideal. Ordering on `name` allows inserting new enum values to be a compatible change, but the user has no control over the desired ordering if they don't want alphabetical. Ordering on `ordinal` gives the user that control, but doesn't allow for inserting additional enum values at a later point without migrating existing data.  Given what Otavio brought up about some of the prominent NoSQL databases having the opposite default as Jakarta Persistence, we will need to define this as being provider-specific. I think it would be fine to state what the default is for Jakarta Persistence or mention the jakarta.persistence.Enumerated annotation that documents what the default is for Jakarta Persistence.  Maybe this would be better done in the JavaDoc.  I can look into that separately.

Reply:
OK, that's fine by me.